### PR TITLE
[76805] Event to ICS Support

### PR DESCRIPTION
**Pull request by mrashed-dev**
# Description
This PR enables support for [generating an ICS file from an Event](https://developer.nylas.com/docs/api/#post/events/to-ics).

# Usage
```java
NylasClient nylas = new NylasClient();
NylasAccount account = nylas.account("{ACCESS_TOKEN}");
Events events = account.events();
Event event = events.get("{eventId}");

// You can make an ICS file from an event ID
String icsFromEventID = events.generateICS(event.getId());

// You can make an ICS file from an Event object
String icsFromEvent = events.generateICS(event);

// You can also pass ICS Options for more configuration
ICSOptions icsOptions = new ICSOptions();
icsOptions.setIcal_uid("test_uuid");
icsOptions.setMethod(ICSMethod.ADD);
icsOptions.setProdid("test_prodid");

String icsFromEventWithOptions = events.generateICS(event, icsOptions);
```

# License
I confirm that this contribution is made under the terms of the MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: src/examples/java/com/nylas/examples/other/EventsExample.java [hunk 1]
```diff
@@ -117,7 +120,9 @@ protected static void basicEventCrud(Events events, Calendar primary) throws IOE
 	}
 
 	protected static void autocreateEvents(Events events, Calendar primary) throws IOException, RequestFailedException {
-		Event created = createBasicEvent(events, primary);
+		Event event = createBasicEvent(events, primary);
+		Event created = events.create(event, true);
+		System.out.println("Created: " + created);
```
Comment (jseller) [inline src/examples/java/com/nylas/examples/other/EventsExample.java:125]: can this use logging? the log4j jar looks like it has been upgraded

Comment (mrashed-dev) [reply src/examples/java/com/nylas/examples/other/EventsExample.java:125]: @jseller good call out! Will update events with logging and re-request your review 😄

Comment (mrashed-dev) [reply src/examples/java/com/nylas/examples/other/EventsExample.java]: @jseller changes made to all examples and merge conflicts resolved.

File: src/examples/java/com/nylas/examples/other/ThreadsExample.java [hunk 2]
```diff
@@ -28,14 +32,14 @@ public static void main(String[] args) throws Exception {
 //				.lastMessageBefore(end)
 				;
 //		List<Thread> allThreads = threads.list(query).chunkSize(10).fetchAll();
-//		System.out.println("result thread count: " + allThreads.size());
+//		log.info("result thread count: " + allThreads.size());
 //		for (Thread thread : allThreads) {
-//			System.out.println(thread);
+//			log.info(thread);
 //		}
```
Comment (jseller) [inline src/examples/java/com/nylas/examples/other/ThreadsExample.java:39]: this is commented out, not sure if that is intentional

File: src/main/java/com/nylas/Events.java [hunk 3]
```diff
@@ -80,6 +80,70 @@ public Event rsvp(String eventId, String status, String accountId, String commen
...
+
+	/**
+	 * Generate an ICS file server-side, from an Event ID
+	 * @param eventId The ID of the existing Event to generate an ICS file for
+	 * @param icsOptions Configuration for the ICS file
+	 * @return String for writing directly into an ICS file
+	 * @see <a href="https://developer.nylas.com/docs/api/#post/events/to-ics">Generate ICS File</a>
+	 */
+	public String generateICS(String eventId, ICSOptions icsOptions) throws RequestFailedException, IOException {
+		if(eventId == null) {
```
Comment (jseller) [inline src/main/java/com/nylas/Events.java:101]: do you have guava or other helper libraries? I think java 11 has isNullorEmpty, or isBlank to check for empty strings

**Review by jseller — APPROVED** [review]
LG! may want to check for the empty string on the event call though

File: src/main/java/com/nylas/Events.java (hunk 3 above)
Comment (mrashed-dev) [reply src/main/java/com/nylas/Events.java]: @jseller thanks! We're targeting Java 8 and I don't want to introduce Guava for just this, but I did forget that we have a `nullOrEmpty` method for strings in our `Validations` class. Using this now, thanks and good call!

**Pull request merged**